launch_testing_ros/launch_testing_ros/wait_for_topics.py

```diff
@@ -50,12 +52,29 @@ def method_2():
             print(wait_for_topics.topics_received()) # Should be {'topic_1', 'topic_2'}
             print(wait_for_topics.messages_received('topic_1')) # Should be [message_1, ...]
             wait_for_topics.shutdown()
+
+        # Method3, calling a callback function before the wait. The callback function takes
+        # the WaitForTopics object as the first argument. Any additional arguments has
+        # to be passed to the wait(*args, **kwargs) method directly.
+        def callback_function(node, arg=""):
+            node.get_logger().info('Callback function called with argument: ' + arg)
+
+        def method_3():
+            topic_list = [('topic_1', String), ('topic_2', String)]
+            wait_for_topics = WaitForTopics(topic_list, timeout=5.0)
+            assert wait_for_topics.wait("Hello World!")
+            print('Given topics are receiving messages !')
+            wait_for_topics.shutdown()
```

Comment:
I'm sure I'm missing something, but this example doesn't really illustrate why the callback function is useful. Could it be modified to do so?

Also, the callback function doesn't seem to actually be provided to `WaitForTopics()`.

Reply:
I modified the docstring so that the callback function is provided to `WaitForTopics()`.

I'm afraid making the example actually useful would be impractical.
As explained in the PR description, the callback is needed when multiple nodes (i.e., multiple publishers and subscribers) are involved.
However, in the unit tests, I try to replicate such conditions as faithfully as possible.